### PR TITLE
Add kyber index exchange adapter

### PR DESCRIPTION
**Pull request by Alpha-Guy**

### CODE REVIEW

File: contracts/protocol/integration/index-exchange/KyberIndexExchangeAdapter.sol [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        uint256 _sourceQuantity,
+        uint256 _destinationQuantity,
+        bytes memory /*_data*/
+    )
+        override
+        external
+        view
+        returns (address, uint256, bytes memory)
+    {
+        KyberTradeInfo memory kyberTradeInfo;
```
Comment (bweick) [inline contracts/protocol/integration/index-exchange/KyberIndexExchangeAdapter.sol:101]: What's the thinking behind putting these variables in a struct?

File: contracts/protocol/integration/index-exchange/KyberIndexExchangeAdapter.sol [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+        kyberTradeInfo.sourceTokenDecimals = ERC20(_sourceToken).decimals();
+        kyberTradeInfo.destinationTokenDecimals = ERC20(_destinationToken).decimals();
+
+        // Get conversion rate from minimum receive token quantity.
```
Comment (bweick) [inline contracts/protocol/integration/index-exchange/KyberIndexExchangeAdapter.sol:106]: what's min receive token? We don't have that defined here

File: contracts/protocol/integration/index-exchange/KyberIndexExchangeAdapter.sol [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            .div(_sourceQuantity);
+
+        // Encode method data for SetToken to invoke
+        bytes memory methodData = abi.encodeWithSignature(
+            "trade(address,uint256,address,address,uint256,uint256,address)",
+            _sourceToken,
+            _sourceQuantity,
+            _destinationToken,
+            _destinationAddress,
+            PreciseUnitMath.maxUint256(), // Sell entire amount of sourceToken
```
Comment (bweick) [inline contracts/protocol/integration/index-exchange/KyberIndexExchangeAdapter.sol:121]: Can you specify a fixed amount to receive with Kyber?

**Comment by bweick** [comment]
What might be nice for adapters is a link to the relevant docs for the external protocol. With perhaps even a short explainer if there's anything weird or different with how it's implemented. This is a bit hard to know exactly what's going on, on the Kyber side.

**Comment by Alpha-Guy** [comment]
Moved to https://github.com/SetProtocol/set-protocol-v2/pull/101

**Pull request closed without merging**